### PR TITLE
Fix naming conflicts in the generated implements mappers

### LINKED ISSUE #477: Resolve naming conflicts in the `FileOpenPicker` class
`IFileOpenPicker`'s [pickSingleFileAsync()](https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.fileopenpicker.picksinglefileasync?view=winrt-22621#windows-storage-pickers-fileopenpicker-picksinglefileasync) conflicts with `IFileOpenPickerWithOperationId`'s [pickSingleFileAsync(string pickerOperationId)](https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.fileopenpicker.picksinglefileasync?view=winrt-22621#windows-storage-pickers-fileopenpicker-picksinglefileasync(system-string)).

### PR DESCRIPTION
**Pull request by halildurmus**
Fixes #477

This PR also excludes some interfaces with deprecated methods and/or properties (`IFileOpenPicker2` and `IFileOpenPickerStatics`).

### CODE REVIEW

File: tool/generator/lib/src/projection/winrt/winrt_class.dart [hunk 1]
```diff
@@ -72,11 +72,18 @@ class WinRTClassProjection extends WinRTInterfaceProjection {
     return buffer.toString();
   }
 
+  /// The WinRT static interfaces to exclude when generating the static mappers.
```
Comment (timsneath) [inline tool/generator/lib/src/projection/winrt/winrt_class.dart]: Does this belong in exclusions.dart?

File: tool/generator/lib/src/projection/winrt/winrt_interface.dart [hunk 2]
```diff
@@ -31,13 +32,26 @@ class WinRTInterfaceProjection extends ComInterfaceProjection {
 
     // This is exposed as dart:core's Duration
     'Windows.Foundation.TimeSpan',
+
+    ...excludedInterfaces,
+  };
+
+  /// The WinRT interfaces to exclude when generating the implements mappers.
+  static const excludedInterfaces = <String>{
```
Comment (timsneath) [inline tool/generator/lib/src/projection/winrt/winrt_interface.dart]: Also in exclusions.dart?

**Review by timsneath — APPROVED** [review]
Perhaps we should keep all exclusions together? WDYT?

File: tool/generator/lib/src/projection/winrt/winrt_class.dart (hunk 1 above)
Comment (halildurmus) [reply tool/generator/lib/src/projection/winrt/winrt_class.dart]: Yes! I moved it to the exclusions.dart file.

File: tool/generator/lib/src/projection/winrt/winrt_interface.dart (hunk 2 above)
Comment (halildurmus) [reply tool/generator/lib/src/projection/winrt/winrt_interface.dart]: Done

**Pull request merged**